### PR TITLE
fix flakey filtered tasks spec

### LINKED ISSUE #13521: [Flaky Test] TaskFilter.filtered_tasks when filtering by case type when filter includes Original and Supplemental and AOD returns tasks with Original or Supplemental case types or AOD cases
## Background/context/resources
```
TaskFilter.filtered_tasks when filtering by case type when filter includes Original and Supplemental and AOD returns tasks with Original or Supplemental case types or AOD cases
Failure/Error: expect(subject.map(&:id)).to match_array((type_1_tasks.map(&:id) + type_2_tasks.map(&:id)) | aod_case_ids)
  expected collection contained:  [215, 216, 217, 218, 219, 220, 221, 224, 227]
  actual collection contained:    [215, 218, 219, 220, 221, 222, 223, 224, 227]
  the missing elements were:      [216, 217]
  the extra elements were:        [222, 223]
# ./spec/models/task_filter_spec.rb:324:in `block (5 levels) in <top (required)>'
```

 - Circle CI Error:  [TaskFilter.filtered_tasks when filtering by case type when filter includes Original and Supplemental and AOD returns tasks with Original or Supplemental case types or AOD cases](https://app.circleci.com/jobs/github/department-of-veterans-affairs/caseflow/104085/parallel-runs/4?filterBy=FAILED)
 - Has the test already been skipped in the code?
   - [ ] Skipped
   - [x] Not Skipped
 - Related Flakes
    + #13470 
    + #13454 

I ran the test locally 100 times with `ensure_stable` and got one failure with the following message:
```
TaskFilter.filtered_tasks when filtering by case type when filter includes Original and upplemental and AOD returns tasks with Original or Supplemental case types or AOD cases
Failure/Error: expect(subject.map(&:id)).to match_array((type_1_tasks.map(&:id) + ype_2_tasks.map(&:id)) | aod_case_ids)
  expected collection contained:  [481, 482, 483, 484, 485, 486, 487, 490, 493]
  actual collection contained:    [483, 486, 487, 488, 489, 490, 491, 492, 495]
  the missing elements were:      [481, 482, 484, 485, 493]
  the extra elements were:        [488, 489, 491, 492, 495]
# ./spec/models/task_filter_spec.rb:325:in `block (6 levels) in <top (required)>'
```

## Approach
Time box this investigation and fix.

Remember that if a test has been skipped for a decent amount of time, it may no longer map to the exact code.

If you reach the end of your time box and don't feel like the solution is in sight:
  - [ ] document the work you've done, including dead ends and research
  - [ ] skip the test in the code
  - [ ] file a follow on ticket
  - [ ] close this issue

### PR DESCRIPTION
**Pull request by tomas-nava**
Resolves #13521 

### Description
This test was flaking when task ids didn't match expected categorizations; this change makes sure that the association is consistent.

### CODE REVIEW

File: spec/models/task_filter_spec.rb [hunk 1]
```diff
@@ -250,35 +250,45 @@
...
           )
         end
       end
 
+      before do
+        create_cached_appeals_for_tasks(tasks_type_original, case_types["1"])
+        create_cached_appeals_for_tasks(tasks_type_supplemental, case_types["2"])
+        create_cached_appeals_for_tasks(tasks_type_post_remand, case_types["3"])
+        create_cached_appeals_for_tasks(tasks_type_reconsideration, case_types["4"])
+        create_cached_appeals_for_tasks(tasks_type_vacate, case_types["5"])
```
Comment (yoomlam) [inline spec/models/task_filter_spec.rb]: To make it more obvious, want to replace `case_type["1"]` with "Original"? Same for other types.
https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/vacols/case.rb#L24

File: spec/models/task_filter_spec.rb [hunk 2]
```diff
@@ -250,35 +250,45 @@
...
-        all_tasks.each_with_index do |task, index|
+        Task.where(
+          id: (
+            tasks_type_original +
+            tasks_type_supplemental +
+            tasks_type_post_remand +
+            tasks_type_reconsideration +
+            tasks_type_vacate
+          ).map(&:id).sort
+        )
```
Comment (yoomlam) [inline spec/models/task_filter_spec.rb]: Out of curiosity, does `where` always return results in the same order as the given params?

File: spec/models/task_filter_spec.rb (hunk 1 above)
Comment (tomas-nava) [reply spec/models/task_filter_spec.rb]: Could do, but I don't mind referring directly to the constants for consistency. That way if the constants ever change, the test will fail and we'll know it has to be updated.

File: spec/models/task_filter_spec.rb (hunk 2 above)
Comment (tomas-nava) [reply spec/models/task_filter_spec.rb]: Good question! I wouldn't rely on it doing so, but it may.

**Comment by tomas-nava** [comment]
> Love the refactor! What was the cause of the flake?

I think that `all_tasks` was occasionally not being sorted as expected, so the following methods that classified the tasks based on their position in that array was off by one or two. As to why the sort was failing, I'm not sure? Maybe a delay in objects being created, so they had nil IDs when the sort was called?

**Pull request merged**